src/psyclone/psyGen.py

```diff
@@ -2781,6 +2803,120 @@ def validate(self, node, options=None):
+                    continue
+                # If the argument is a keyword argument, i.e. it has a default
+                # value then we add it to the list of options.
+                if v.default is not inspect.Parameter.empty:
+                    default = v.default
+                else:
+                    # If its not a keyword argument then we skip it.
+                    continue
+                # If there is type information on the keyword argument, then we
+                # store the information so we can generate docs and do
+                # automatic type checking of options.
+                if v.annotation is not inspect.Parameter.empty:
+                    type_anno = v.annotation
+                    typename = stringify_annotation(
+                            v.annotation
+                    )
+                else:
+                    type_anno = None
+                    typename = None
+                valid_options[k] = ValidOption(
+                    default=default, type=type_anno, typename=typename
+                )
+        return valid_options
+
+    def validate_options(self, **kwargs):
+        '''
+        Checks the options provided to this transformation are valid. Makes
+        use of get_valid_options to work out the available options.
+
+        :raises ValueError: If an invalid option was provided.
+        :raises TypeError: If an option was provided with a value of the wrong
+                           type.
+        '''
+        valid_options = type(self).get_valid_options()
+        invalid_options = []
+        wrong_types = {}
+        for option in kwargs:
+            if option == "options":
+                continue
+            if option not in valid_options:
+                invalid_options.append(option)
+                continue
+            if valid_options[option].type is not None:
+                try:
+                    if not isinstance(
+                            kwargs[option], valid_options[option].type):
+                        wrong_types[option] = type(kwargs[option]).__name__
+                except TypeError:
+                    # For older versions of Python, such as 3.8 they don't yet
+                    # support type checking for Generics, e.g. Union[...] so
+                    # we skip this check and it needs to be done in the
+                    # relevant function instead.
+                    pass
+
+        if len(invalid_options) > 0:
+            invalid_options_detail = []
+            for invalid in invalid_options:
+                invalid_options_detail.append(f"'{invalid}'")
+            invalid_options_list = ", ".join(invalid_options_detail)
+            raise ValueError(f"'{type(self).__name__}' received invalid "
+                             f"options [{invalid_options_list}]. Please see "
+                             f"the documentation and check the available "
+                             f"options.")
```

Comment:
we could just print the list of valid options here too.

Reply:
Added